File: HorseRacing.Contracts/Models/Game/Requests/JoinGameRequest.cs
using HorseRacing.Contracts.Base.Requests;
using HorseRacing.Contracts.Models.Game.Dtos;

namespace HorseRacing.Contracts.Models.Game.Requests
{
    public class JoinGameRequest : BaseRequest<JoinGameRequestDto>
    {
        public JoinGameRequest() : base() { }
        public JoinGameRequest(JoinGameRequestDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/game/requests/join-game/join-game-request.ts
import type { BaseRequest } from "~/interfaces/api/contracts/base/requests/base-request";
import type { JoinGameRequestDto } from "./join-game-request-dto";

export class JoinGameRequest implements BaseRequest<JoinGameRequestDto> {
    Data: JoinGameRequestDto;

    constructor(data?: JoinGameRequestDto) {
        this.Data = data || { UserId: '', GameId: '' };
    }
}
